### PR TITLE
threatgrid/ctim/#381 Migrate actor 1.2.0

### PR DESCRIPTION
**Pull request by samwagg**
Related threatgrid/ctim#381

Update actor to reflect new CTIM model, in which actor_types is modified to be both optional and an array. Add migration code.

I ran the migration locally following the directions in the migration.md doc and it worked fine. I also confirmed that I was able to search by actor_types afterward.

<a name="iroh-services-clients">[§](#iroh-services-clients)</a> IROH Services Clients
=====================================================================================

**Not sure if something is needed here.**

<a name="qa">[§](#qa)</a> QA
============================

Describe the steps to test your PR.

1. Only testable in private CTIA because public CTIA doesn't have any loaded Actors.
2. Check that `actor_type` is gone and replaced with `actor_types`, containing a string array (with only a single element for now) for all Actors
3. Check that search by `actor_types` works correctly.

<a name="ops">[§](#ops)</a> Ops
===============================

  Specify Ops related issues and documentation
- Migration and config change needed: https://github.com/advthreat/tenzin-config/pull/791

<a name="documentation">[§](#documentation)</a> Documentation
=============================================================

  Public Facing documentation section;
**Not sure if anything is needed for public documentation**

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================
```
intern: changes to actor schema including `actor_type` -> `actor_types`
```

### CODE REVIEW

File: src/ctia/task/migration/migrations.clj [hunk 1]
```diff
@@ -166,6 +175,7 @@
    :1.0.0 (comp (append-version "1.0.0")
                 (rename-observable-type "pki-serial" "pki_serial")
                 simplify-incident)
+   :1.2.0 actor-type-array
```
Comment (ereteog) [inline src/ctia/task/migration/migrations.clj:178]:
```suggestion
   :1.2.0 (comp (append-version "1.2.0") actor-type-array)
```

**Review by ereteog — COMMENTED** [review]
it looks good but could you include the new schema-version in the transformation and test the schema bump, please?

**Comment by samwagg** [comment]
> it looks good but could you include the new schema-version in the transformation and test the schema bump, please?

Done. I'm also working on a PR for tenzin-config right now, just fyi.

**Comment by samwagg** [comment]
Here is the tenzin-config PR: https://github.com/advthreat/tenzin-config/pull/791

**Review by ereteog — APPROVED** [review]
LGTM

**Comment by trcollier** [comment]
QA verified in TEST release v1.112

**Comment by trcollier** [comment]
QA verified in TEST release v1.112

**Pull request merged**